akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -12,7 +16,11 @@ import scala.util.{ Failure, Success, Try }
  * @param count Numeric value depending on context, for example IO operations performed or bytes processed.
  * @param status Status of the result. Can be either [[akka.Done]] or an exception.
  */
-final case class IOResult private[stream] (count: Long, status: Try[Done]) {
+final case class IOResult private (count: Long, status: Try[Done])
+  extends Serializable with Product {
```

Comment:
True, however this reminds me of a problem that we get once it's a case class again - copy method which will break if we do anything to this class in the future. Perhaps no matter heh, those we'll have to evolve more painfully then anyway